### PR TITLE
refactor: make workflows independent and minimize the unnecessary runs

### PR DESCRIPTION
**Pull request by pratikjagrut**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #
This PR minimizes the unnecessary runs of ci.

### CODE REVIEW

File: .github/workflows/unit-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Unit test
+
+on:
+  release:
+    types: [created]
+  pull_request:
+    branches:
+      - main
```
Comment (FabianKramm) [inline .github/workflows/unit-tests.yaml:8]: we still use master branch, so I guess we need to change that here

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Lint
+
+on:
+  release:
+    types: [created]
+  pull_request:
+    branches:
+      - main
```
Comment (FabianKramm) [inline .github/workflows/lint.yaml:8]: we still use master branch, so I guess we need to change that here

File: .github/workflows/e2e-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+name: E2E tests
+
+on:
+  release:
+    types: [created]
+  pull_request:
+    branches:
+      - main
```
Comment (FabianKramm) [inline .github/workflows/e2e-tests.yaml:8]: we still use master branch, so I guess we need to change that here

File: .github/workflows/unit-tests.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          go-version: 1.17
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v1
+
+      - name: Test
+        run: ./hack/coverage.bash
+
+  test-unit-windows:
+    if: github.ref != 'refs/heads/master'
```
Comment (FabianKramm) [inline .github/workflows/unit-tests.yaml:40]: why does we have this for windows but not for the others?

File: .github/workflows/pr-checks.yml [hunk 5]
```diff
@@ -1,92 +0,0 @@
-name: PR checks
-
-on: 
-  pull_request:
-    branches: [ master ]
-
-env:
```
Comment (FabianKramm) [inline .github/workflows/pr-checks.yml:7]: are these not needed anymore if we split up those checks?

**Review by FabianKramm — CHANGES_REQUESTED** [review]
@pratikjagrut thanks for the PR! Please see my comments

File: .github/workflows/unit-tests.yaml (hunk 4 above)
Comment (pratikjagrut) [reply .github/workflows/unit-tests.yaml:40]: I guess this is a mistake. We don't need any condition here.

File: .github/workflows/unit-tests.yaml (hunk 1 above)
Comment (pratikjagrut) [reply .github/workflows/unit-tests.yaml:8]: ah yes.

File: .github/workflows/pr-checks.yml (hunk 5 above)
Comment (pratikjagrut) [reply .github/workflows/pr-checks.yml:7]: Yes, we don't need this file.

**Review by FabianKramm — APPROVED** [review]
thanks @pratikjagrut ! LGTM!

**Pull request merged**